packages/core/contracts/insured-bridge/BridgePool.sol

```diff
@@ -388,13 +383,60 @@ contract BridgePool is Testable, BridgePoolInterface, ExpandedERC20, MultiCaller
         updateAccumulatedLpFees();
         allocateLpFees(_getAmountFromPct(relay.realizedLpFeePct, _depositData.amount));
 
-        emit RelaySettled(depositHash, relayHash, msg.sender);
+        emit RelaySettled(depositHash, _getRelayHash(_depositData, relay), msg.sender);
 
         delete instantRelays[keccak256(abi.encode(depositHash, relay.realizedLpFeePct))];
         delete relay.realizedLpFeePct;
         delete relay.priceRequestTime;
     }
 
+    /**
+     * @notice Callback for disputes, marks relay as disputed.
+     * @dev Reverts if relay is not pending.
+     * @dev _timestamp and _identifier are unused because _ancillaryData contains a relay nonce and uniquely
+     * identifies a relay request.
+     * @param _identifier price identifier for relay request.
+     * @param _timestamp timestamp for relay request.
+     * @param _ancillaryData ancillary data for relay request.
+     * @param _request disputed relay request params.
+     */
+    function priceDisputed(
+        bytes32 _identifier,
+        uint32 _timestamp,
+        bytes memory _ancillaryData,
+        SkinnyOptimisticOracleInterface.Request memory _request
+    ) external onlyFromOptimisticOracle {
+        bytes32 depositHash = relayRequestAncillaryData[_ancillaryData];
+        RelayData storage relay = relays[depositHash];
+        require(relay.relayState == RelayState.Pending);
```

Comment:
What happens if this require fails? Just the relay state won't get set?

Reply:
it wont impact the OO as it's within a try catch that side. It would mean the relay remains in `Pending` State I guess.

Reply:
Yeah I forget exactly what the situation was that prompted me to add this `require`, but the following `require(relay.relayState == RelayState.Pending || relay.relayState == RelayState.Disputed)` in `priceSettled` addresses the following situation:

- Relay request 1 is sent --> `RelayState = Pending`
- Someone disputes relay request 1 --> `RelayState = Disputed`
- Relay request 2 is sent with same params as request 1 --> `RelayState = Pending`
- Request 2 passes challenge period --> `RelayState = Pending`
- OO settles price request originating from request 2 --> `RelayState = PendingFinalization
- Someone calls `settleRelay` on BridgePool --> `RelayState = Finalized`
- Someone resolves dispute of relay request 1 

Without the above `require`, this last settlement would reset the `RelayState` to `PendingFinalization` and allow it to be settled again. It's possible because the following code in `priceSettled` would point to the same `relay` (that has already been Finalized!):

```
function priceSettled(
        bytes32 _identifier,
        uint32 _timestamp,
        bytes memory _ancillaryData,
        SkinnyOptimisticOracleInterface.Request memory _request
    ) external onlyFromOptimisticOracle {
        bytes32 depositHash = relayRequestAncillaryData[_ancillaryData];
        RelayData storage relay = relays[depositHash];
```

Reply:
@chrismaree @mrice32 can you let me know if you agree with the above logic? The require statement that Matt mentioned I think can be removed, do you see a situation where removing it creates a case where a RelayState gets stuck?

Reply:
ye, I think the flow you pointed out makes sence and having the modifier on the `priceSettled` callback is sufficient for this. I agree without the modifier we'd have issues with re-setting this state to.

I cant right now think of a case where this will get stuck (I'll spend some more time looking at this particular bit of code though).

One really nice sideffect of this change is the logic is really really simple now: only if it's in `RelayState.PendingFinalization` can it be settled.

Please can you add a comment above the require explaining how this works in the re-relayed case to make it clear for us in the future & to help with audit?

Reply:
I'm going to change this logic a lot in the next PR so I'm going to skip comments for now